src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void handleOperation(FruitTransaction fruitTransaction, Storage storage) {
+        storage.getInventory().merge(
+                fruitTransaction.getFruit(),
+                fruitTransaction.getQuantity(),
+                Integer::sum
+        );
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Currently, the `handleOperation` method incorrectly increases the quantity of fruit in the storage, which is not the expected behavior for a purchase operation. Instead, it should decrease the quantity and ensure that the result does not lead to a negative balance.